storage/container_storage_interface/persistent-storage-csi-azure-file.adoc

```diff
@@ -0,0 +1,40 @@
+[id="persistent-storage-csi-azure-file"]
+= Azure File CSI Driver Operator
+include::modules/common-attributes.adoc[]
+:context: persistent-storage-csi-azure-file
+
+toc::[]
+
+== Overview
+
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Microsoft Azure File Storage.
+
+:FeatureName: Azure File CSI Driver Operator
+include::modules/technology-preview.adoc[leveloffset=+1]
+
+Familiarity with xref:../../storage/understanding-persistent-storage.adoc#understanding-persistent-storage[persistent storage] and xref:../../storage/container_storage_interface/persistent-storage-csi.adoc#persistent-storage-csi[configuring CSI volumes] is recommended when working with a CSI Operator and driver.
+
+To create CSI-provisioned PVs that mount to Azure File storage assets with this feature is enabled, {product-title} installs the Azure File CSI Driver Operator and the Azure File CSI driver by default in the `openshift-cluster-csi-drivers` namespace.
+
+* The _Azure File CSI Driver Operator_, after being enabled, provides a storage class named `azurefile-csi` that you can use to create persistent volume claims (PVCs). The Azure File CSI Driver Operator supports dynamic volume provisioning by allowing storage volumes to be created on-demand, eliminating the need for cluster administrators to pre-provision storage.
+
+* The _Azure File CSI driver_ enables you to create and mount Azure File PVs.
+
+include::modules/persistent-storage-csi-about.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+{product-title} defaults to using an in-tree (non-CSI) plug-in to provision Azure File storage.
+
+In future {product-title} versions, volumes provisioned using existing in-tree plug-ins are planned for migration to their equivalent CSI driver. CSI automatic migration should be seamless. Migration does not change how you use all existing API objects, such as persistent volumes, persistent volume claims, and storage classes. For more information about migration, see xref:../../storage/container_storage_interface/persistent-storage-csi-migration.adoc#persistent-storage-csi-migration[CSI automatic migration].
```

Comment:
We tend to avoid making promises for the future in the docs. If it's vitally important to include this note, maybe run it past Vikram first.